Source/Flow/Private/FlowAsset.cpp

```diff
@@ -90,74 +124,124 @@ EDataValidationResult UFlowAsset::ValidateAsset(FFlowMessageLog& MessageLog)
 				MessageLog.Messages.Append(Node.Value->ValidationLog.Messages);
 			}
 		}
+		else
+		{
+			const FString ErrorMsg = FString::Format(*ValidationError_NullNodeInstance, {*Node.Key.ToString()});
+			MessageLog.Error(*ErrorMsg, this);
+		}
 	}
 
 	return MessageLog.Messages.Num() > 0 ? EDataValidationResult::Invalid : EDataValidationResult::Valid;
 }
 
-bool UFlowAsset::IsNodeClassAllowed(const UClass* FlowNodeClass) const
+bool UFlowAsset::IsNodeClassAllowed(const UClass* FlowNodeClass, FText* OutOptionalFailureReason) const
 {
-	if (FlowNodeClass == nullptr)
+	if (!IsValid(FlowNodeClass))
+	{
+		return false;
+	}
+
+	if (!CanFlowNodeClassBeUsedByFlowAsset(*FlowNodeClass))
```

Comment:
These checks confirm access to the FlowNode is legal for the plugins that this flow graph depends-on.